snuba/query/parser/__init__.py

```diff
@@ -6,7 +6,9 @@
 from snuba import environment
 from snuba.clickhouse.escaping import NEGATE_RE
 from snuba.datasets.dataset import Dataset
+from snuba.datasets.entities.factory import get_entity
 from snuba.datasets.entity import Entity
+from snuba.query.data_source.simple import Entity as QueryEntity
```

Comment:
The naming of this class seems like we might be heading towards the whole `FunctionCall as FunctionCallMatch` fiasco. I'm not saying you need to change the class name, just pointing it out for consideration.

Reply:
Yeah, actually I thought about that a bit. In that case the fiasco was because we were pretty much always using expressions and matchers in the same files (all the time) so they were always both imported. 
It believe it should not be the case here, since query and data source are mostly used in query processing logic (processors, translators, etc.), which cannot import the structural entity class from the dataset (that always end up being a circular dependency), so importing both should be only a case in the parser.